### DISCUSSION TITLE
Ps4 controller setup

### DISCUSSION THREAD

**Discussion by Jopp-gh** [AntiMicroX/antimicrox#1056] [Q&A]
OS: Linux mint, 21
Version: AntiMicroX 3.4.1, 
package: Appimage

Hello, my first post here🙂
Since I didn't find any clear info, how to get my ps4 controller working, I figured by trial and error, here are my installations:
- xboxdrv (probably essential)
- ds4drv (translates xboxdrv for ps4 controllers)
- game pad config tool (reorders buttons and axes)
- AntiMicroX 

I wonder why I've to keep all these packages, to get my ps4 controller working. Ideally, AntiMicroX should be enough and maybe a running xboxdrv background process, listening for controller connections. But, to be honest, I just need a working controller for games, so a bg process isn't that important. 

Let me explain, ds4drv is a very old and abandoned PlayStation driver, so I'd prefer to replace it with AntiMicroX, but this ds4drv seems essential, otherwise the order of buttons and axes are messed up and there's zero signal transmission on my ps4 controller 

It seems, AntiMicroX was made exclusively for Xbox controllers. No real critique to AntiMicroX because it works well and allows all kind of configuration freedom, but PlayStation controllers and Xbox controllers are the only gamepads nowadays. 

My question, are you planning to add PlayStation support as well and if yes, are you interested to integrate PlayStation controller support as translation layer of "xboxdrv" into AntiMicroX?

**Comment by pktiuk** [comment]
As far as I know Current Linux kernel has already implemented PS4 Gamepad drivers in the kernel.     
You don't need any app to use your gamepad (if your game is compatible with gamepads)

On my freshly installed Fedora 40, I just connected my PS4 controller and AntiMicroX recognizes it without any problem.

**Reply by Jopp-gh** [reply]
Good news!
Tomorrow I'll run yet another test, uninstall older drivers and see if AntiMicroX works alone too. Would be a relief.

Probably I've just to delete `~/.local/share/gamecontrollerdb.local.text` generated by "game pad config tool"(only for Xbox gamepads) to get additional keys on my ps4 controller unlocked.

**Reply by Jopp-gh** [reply]
I uninstalled **all old drivers** and AntiMicroX still works, that's Awesome! I hope this will stay like that even after a reboot.

Two things are still missing, my controller isn't yet seen as ps4 device and re-assigning, re-ordering axes and buttons didn't help. 

On my ps4 I've one more button to use. It's not much, but a mapping more is always an advantage.

In the same window,
I would prefer to see my ps4 gamepad as guide image instead the Xbox gamepad guide image.
